src/pages/settings/CurrentProjectSelect.tsx

```diff
@@ -0,0 +1,144 @@
+  const { isRestricted } = useAuth();
+  const toastNotify = useToastNotification();
+
+  const configField = {
+    key: "user.ui_current_project",
+    default: "",
+  };
+
+  const projectOptions = [
+    {
+      label: "All projects",
+      value: ALL_PROJECTS,
+    },
+    ...projects.map((project) => ({
+      label: project.name,
+      value: project.name,
+    })),
+  ];
+
+  const canBeReset = String(configField.default) !== String(value);
+
+  const handleSave = () => {
+    saveCurrentProject(value);
+
+    const settingLabel = (
+      <ResourceLabel bold type="setting" value={configField.key} />
+    );
+    toastNotify.success(<>Setting {settingLabel} updated.</>);
+    setEditMode(false);
+  };
+
+  const resetToDefault = () => {
+    setValue(configField.default || "");
+  };
+
+  const onCancel = () => {
+    setValue(loadCurrentProject() || "");
+    setEditMode(false);
+  };
+
+  useEffect(() => {
+    if (!isEditMode) {
+      setValue(loadCurrentProject() || "");
+      return;
+    }
+    setValue(getDefaultProject(projects));
+  }, [isEditMode, projects]);
+
+  return (
+    <>
+      {isEditMode && (
+        <Form
+          onSubmit={(e) => {
+            e.preventDefault();
+            handleSave();
+          }}
+        >
+          <Select
+            name={`${configField.key}-select`}
+            aria-label={configField.key}
+            options={projectOptions}
+            value={value}
+            onChange={(e) => {
+              setValue((e.target as HTMLSelectElement).value);
+            }}
+          />
+          <Button appearance="base" onClick={onCancel}>
+            Cancel
+          </Button>
+          <Button appearance="positive" type="submit">
+            Save
+          </Button>
+          {canBeReset && (
+            <Button
+              className="reset-button"
+              appearance="base"
+              onClick={resetToDefault}
+              hasIcon
+            >
+              <Icon name="restart" className="flip-horizontally" />
+              <span>Reset to default</span>
+            </Button>
+          )}
+        </Form>
+      )}
+      {!isEditMode && (
+        <>
+          {isRestricted ? (
+            <span>
+              {value === ALL_PROJECTS ? "All projects" : (value ?? "-")}
+            </span>
+          ) : (
+            <Button
+              appearance="base"
+              className="readmode-button u-no-margin"
+              onClick={() => {
+                setEditMode(true);
+              }}
+              hasIcon
+              disabled={!canEditServerConfiguration()}
+              title={
+                canEditServerConfiguration()
+                  ? ""
+                  : "You do not have permission to edit server configuration"
+              }
+            >
+              <div className="readmode-value u-truncate">
+                {value === ALL_PROJECTS ? "All projects" : value || "-"}
```

Comment:
Instead of falling back to "-" I think we should fall back to the new extracted function.